packages/patternfly-3/patternfly-react/src/components/DateTimePicker/DateTimePicker.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl, InputGroup, Icon, OverlayTrigger, Popover } from '../../';
+import DateInput from './DateComponents/DateInput';
+import TodayButton from './DateComponents/TodayButton';
+import TimeInput from './TimeComponents/TimeInput';
+
+class DateTimePicker extends React.Component {
+  state = {
+    value: new Date(this.props.value),
+    typeOfDateInput: 'M',
+    isTimeTableOpen: false
+  };
+  formatDate = () => {
+    const { locale } = this.props;
+    const { value } = this.state;
+    const options = [{ year: 'numeric', month: 'numeric', day: 'numeric' }, { hour: '2-digit', minute: '2-digit' }];
+    return `${value.toLocaleString(locale, options[0])} ${value.toLocaleString(locale, options[1])}`;
+  };
+  setSelected = date => {
+    if (Date.parse(date)) {
+      const newDate = new Date(date);
+      this.setState({ value: newDate });
+    }
+    this.setState({
+      typeOfDateInput: 'M',
+      isTimeTableOpen: false
+    });
+  };
+  render() {
+    const { locale, weekStartsOn, inputProps } = this.props;
+    const { value, typeOfDateInput, isTimeTableOpen } = this.state;
+    const popover = (
+      <Popover id="popover-date-picker" className="bootstrap-datetimepicker-widget dropdown-menu timepicker-sbs">
+        <div className="row">
+          <DateInput
+            date={value}
+            setSelected={this.setSelected}
+            locale={locale}
+            weekStartsOn={weekStartsOn}
+            className="col-md-6"
+            typeOfDateInput={typeOfDateInput}
+          />
+          <TimeInput time={value} setSelected={this.setSelected} isTimeTableOpen={isTimeTableOpen} />
+        </div>
+        <li className="picker-switch accordion-toggle">
+          <TodayButton setSelected={this.setSelected} />
+        </li>
+      </Popover>
+    );
+    return (
+      <div>
+        <OverlayTrigger trigger="click" placement="top" overlay={popover} rootClose>
+          <InputGroup className="input-group date-time-picker-pf">
+            <FormControl
+              {...inputProps}
+              aria-label="date-picker-input"
+              type="text"
+              value={this.formatDate()}
```

Comment:
Date Time input value should be [controlled](https://reactjs.org/docs/forms.html) and support nul (doesn't have to be required in forms)